File: ide/src/main/java/org/dau/ui/fx/FxCleaner.java
package org.dau.ui.fx;

import javafx.application.Platform;

import java.lang.ref.Cleaner;

public final class FxCleaner {

  private static final Cleaner CLEANER = Cleaner.create();

  public static Cleaner.Cleanable clean(Object ref, Runnable runnable) {
    return CLEANER.register(ref, () -> Platform.runLater(runnable));
  }
}

File: ide/src/main/kotlin/org/dau/ui/fx/FxCleaner.kt
package org.dau.ui.fx

import javafx.application.Platform
import java.lang.ref.Cleaner
import java.lang.ref.Cleaner.Cleanable

object FxCleaner {

    private val CLEANER = Cleaner.create()

    @JvmStatic
    fun clean(ref: Any?, runnable: Runnable?): Cleanable {
        return CLEANER.register(ref) { Platform.runLater(runnable) }
    }
}